src/main/java/org/osiam/client/AuthService.java

```diff
@@ -333,6 +496,12 @@ private void checkAndHandleResponse(String content, StatusType status, AccessTok
         } else if (status.getStatusCode() == Status.FORBIDDEN.getStatusCode()) {
             String errorMessage = extractErrorMessageForbidden(accessToken);
             throw new ForbiddenException(errorMessage);
+        } else if (status.getStatusCode() == Status.NOT_FOUND.getStatusCode()) {
```

Comment:
I think we should get rid of this huge else if block. (It's almost a case statement.) using object oriented design and polymorphism could be the way to go.